modules/auxiliary/scanner/http/git_scanner.rb

```diff
@@ -0,0 +1,75 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::WmapScanServer
+  include Msf::Auxiliary::Scanner
+  include Msf::Auxiliary::Report
+
+  def initialize
+    super(
+      'Name'        => 'HTTP Git Scanner',
+      'Description' => 'Detect Git disclosure.',
+      'Author'      => ['Nixawk'],
+      'References'  => [
+        ['URL', 'https://github.com/git/git/blob/master/Documentation/technical/index-format.txt']
+      ],
+      'License'     => MSF_LICENSE
+    )
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The test path to .git directory', '/'])
+      ], self.class)
+  end
+
+  def peer
+    "#{rhost}:#{rport}"
+  end
+
+  def git_url(url)
+    normalize_uri(url, '/.git/index')
+  end
+
+  def git_index_parse(resp)
+    return if resp.blank? || resp.length < 12 # A 12-byte header
+    standard_signature = 'DIRC'
```

Comment:
This could just be made a constant -- rename it to `GIT_SIG` and stuff it outside of a method inside this class.